Realm/ObjectStore/shared_realm.hpp

```diff
@@ -101,15 +107,33 @@ namespace realm {
         void verify_thread() const;
         void verify_in_write() const;
 
+        bool can_deliver_notifications() const noexcept;
+
         // Close this Realm and remove it from the cache. Continuing to use a
         // Realm after closing it will produce undefined behavior.
         void close();
 
         ~Realm();
 
-      private:
+        void init(std::shared_ptr<_impl::RealmCoordinator> coordinator);
         Realm(Config config);
 
+        // Give AsyncQuery direct access to the shared group as it needs it to
```

Comment:
The comment mentions only `AsyncQuery` but the code also gives access to `RealmCoordinator`.